exporter/src/main/scala/uk/gov/nationalarchives/consignmentexport/BagMetadata.scala

```diff
@@ -0,0 +1,72 @@
+package uk.gov.nationalarchives.consignmentexport
+
+import java.util.UUID
+
+import cats.effect.IO
+import gov.loc.repository.bagit.domain.Metadata
+import graphql.codegen.GetConsignmentExport
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import uk.gov.nationalarchives.consignmentexport.BagMetadata._
+import uk.gov.nationalarchives.consignmentexport.Config.{Configuration, config}
+
+class BagMetadata(graphQlApi: GraphQlApi, keycloakClient: KeycloakClient)(implicit val logger: SelfAwareStructuredLogger[IO]) {
+  def getBagMetadata(consignmentId: UUID, config: Configuration): IO[Metadata] = for {
+   consignment <- graphQlApi.getConsignmentExport(config, consignmentId)
+   metadata = new Metadata
+
+   //Question throw exception if fail to add consignment info?
+   consignmentInfoAdded = consignment match {
+     case Some(consignment) => addConsignmentMetadata(consignment, metadata)
+     case None => false
+   }
+
+  } yield metadata
+
+  private def addConsignmentMetadata(consignment: GetConsignmentExport.getConsignmentExport.GetConsignment, metadata: Metadata): Boolean = {
+    //Question if a value is missing from the consignment info, should a default value be added, or ignore, or throw exeception?
+    for {
+      series <- consignment.series
+      sc <- series.code
+    } yield metadata.add(ConsignmentSeriesKey, sc)
```

Comment:
I haven't looked at the tests yet, but given that you're dealing with the mutable third-party Metadata class here, I'm guessing it will be much easier to test if you make this method return an object containing all the metadata, and then add it to the bag as a separate step?

Reply:
Are you suggesting that I use a case class to hold the properties from the metadata, returning this case class and use this to populate the bag `Metadata` object?

Reply:
Yeah, a list of instance of a case class, or a Map. Whatever nice immutable data structure works.